0100-private-projects.md

```diff
@@ -0,0 +1,36 @@
+- Feature Name: Private Projects
+- Created: 2020-02-10
+- Associated Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1602490
+
+# Summary
+
+Add an option to make a project "private", restricting who can see and interact with it.
+
+# Motivation
+
+It can be useful in some cases to be able to hide a project from users. For example, when a superuser creates a new project, they might want to make sure that everything is correctly set up before exposing it — running sync, verifying that it is enabled for the right locales, etc. In such a case, the project being not visible from non-superusers is enough. Once the superuser is done with configuration, they simply make the project public.
+
+For a different use-case, let's consider an agency working with several clients. They have a number of projects created on Pontoon, and they want to have specific localizers working on specific projects. It would be beneficial for them to able to restrict seeing a project to only a defined list of users. This requires being able to mark a project as private, and also having a configuration option to select a list of users who will be able to see, and interact, with that project. That would allow a clear separation of projects, reducing the risk of localizers translating the wrong set of projects.
+
+# Feature explanation
+
+When creating or editing a project, a superadmin has two new options:
+
+- a simple checkbox to make the project private or public;
+- a formset controlling which users can see the project when it is private.
+
+The "approved users" formset is visible only when the project is marked as private. It has two elements: a list of users, showing who currently has access (or will upon saving), and a search input, used to look for a specific user in the database and adding that user to the list. Clicking a user in the list removes that user from the list.
```

Comment:
Our Pontoon instance has a massive amount of users, most of which are dormant. We probably can't expose "everyone" on that list?

Reply:
Yes, that's why there are two elements to the UI: a list of currently approved users (which should be small), and an input to look for a specific user in the DB. It's never showing the full list of all users.